### PR TITLE
Prevent update of a MachineDeployment after it is initially created

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
MachineDeployments are updated by CAPI after they are created to apply
defaults. This change sets the defaults on initial reconcile to avoid
an unnecessary update.

**Checklist**
- [x] Subject and description added to both, commit and PR.

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -975,7 +984,7 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 		// Before persisting, if the NodePool is brand new we want to make sure the replica number is set so the machineDeployment controller
 		// does not panic.
 		if machineDeployment.Spec.Replicas == nil {
-			machineDeployment.Spec.Replicas = k8sutilspointer.Int32Ptr(k8sutilspointer.Int32PtrDerefOr(nodePool.Spec.Replicas, 0))
+			setMachineDeploymentReplicas(nodePool, machineDeployment)
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:987]: to clarify, is this change cosmetic or does it also intend to have an impact to reduce generation?

**Comment by alvaroaleman** [comment]
/hold
verify failure won't recover itself

File: hypershift-operator/controllers/nodepool/nodepool_controller.go (hunk 1 above)
Comment (csrwng) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:986]: It does have an impact in that for autoscaling nodepools the replicas would first be set to 0, then when this block was skipped they would get set to 1 by `setMachineDeploymentReplicas` resulting in another generation.

**Review by alvaroaleman — APPROVED** [review]
/hold cancel

**Pull request merged**